scripts/spack/packages/axom/package.py

```diff
@@ -131,8 +131,8 @@ class Axom(CMakePackage, CudaPackage):
         depends_on('umpire cuda_arch={0}'.format(sm_),
                    when='+umpire cuda_arch={0}'.format(sm_))
 
-    depends_on("mfem", when="+mfem")
-    depends_on("mfem~mpi", when="+mfem~mpi")
+    depends_on("mfem+mpi+metis+zlib", when="+mfem+mpi")
```

Comment:
Moving these out of the packages.yaml file will force the metis and zlib variants on our downstream customers.  Are you sure you meant to do this?

Reply:
>      depends_on("mfem+mpi+metis+zlib", when="+mfem+mpi") 

It's likely that I got this wrong.

I wanted to make axom's `mfem` be parallel when axom is built with `mpi`, but be serial otherwise.
Parallel MFEM depends on hypre and metis (and presumably zlib? not sure)

In the current setup,   `mfem` defaults to `mfem+mpi+metis+zlib`, so I can probably simplify this to the following:
```diff
- depends_on("mfem+mpi+metis+zlib", when="+mfem+mpi")
- depends_on("mfem~mpi~metis~zlib", when="+mfem~mpi")
+ depends_on("mfem", when="+mfem") 
+ depends_on("mfem~mpi~metis~zlib", when="+mfem~mpi")
```

See: 
https://github.com/LLNL/axom/blob/2c7b51a59c1e9e5e1e34cdc8b7f4e7b1a64a9cb5/scripts/spack/packages/mfem/package.py#L94-L98
https://github.com/LLNL/axom/blob/2c7b51a59c1e9e5e1e34cdc8b7f4e7b1a64a9cb5/scripts/spack/packages/mfem/package.py#L138-L139

Is that right?

Reply:
This PR keeps our mfem serial, so this discussion is no longer relevant.  
I left the partial solution in the git history as breadcrumbs for when I revisit.